mtls-southbound/README.md

```diff
@@ -0,0 +1,146 @@
+```
+
+## Step 2: Create a VM with nginx using mTLS
+
+If you already have an mTLS endpoint with certificate and key, you can skip this step. First we will create two firewall rules to allow traffic to the VM, and then we will create a small VM with [nginx](https://nginx.org/) running to handle the mTLS backend requests.
+
+```sh
+# create firewall rules for port 22.
+gcloud compute firewall-rules create allow22 --allow tcp:22 --project $PROJECT_ID --target-tags allow22
+
+# create our backend VM, with a startup script to install nginx and let us write files to the /etc/nginx dir.
+gcloud compute instances create $VM_NAME \
+  --project=$PROJECT_ID \
+  --zone=$ZONE \
+  --image=debian-12-bookworm-v20250610 \
+  --image-project=debian-cloud \
+  --machine-type=e2-medium \
+  --tags=https-server,allow22 \
+  --metadata=startup-script='#! /bin/bash
+apt update
+apt -y install nginx
+sudo chmod -R 777 /etc/nginx'
+```
+
+Now set the environment variable `export VM_IP=YOUR_VM_EXTERNAL_IP` with the EXTERNAL IP that is displayed in the VM create output, or update and source your `.env` file if you created one.
+
+## Step 3: Create self-signed certificate and key
+
+Now we will create a self-signed certificate and key to test with.
+
+```sh
+# create cert.pem certificate and set to allow our VM_IP as Alt.
+openssl req -subj '/CN=ssl.test.local' -x509 -new -newkey rsa:4096 -keyout key.pem -out cert.pem -sha256 -days 365 -nodes -addext "keyUsage = digitalSignature,keyAgreement" -addext "extendedKeyUsage = serverAuth, clientAuth" -addext "subjectAltName = DNS:ssl.test.local, DNS:localhost, IP:127.0.0.1, IP:$VM_IP"
+
+# create PCKS12 export, use 'test' as password
+openssl pkcs12 -export -out client.p12 -inkey key.pem -in cert.pem
+
+# verify certificate, should return OK
+openssl verify -CAfile cert.pem cert.pem
+```
+
+## Step 4: Set VM cert and nginx config
+
+Now we will sync the cert and key to the VM, and set the nginx config file as well.
+
+```sh
+# copy cert, key and nginx.conf files to VM /etc/nginx dir
+gcloud compute scp cert.pem key.pem nginx.conf $VM_NAME:/etc/nginx --zone=$ZONE --project $PROJECT_ID
+
+# restart nginx to apply config
+gcloud compute ssh $VM_NAME --zone=$ZONE --project=$PROJECT_ID --command="sudo nginx -s reload"
+```
+
+## Step 5: Test calling VM directly with cert and key
+
+Now that our nginx is running with our cert and key, we can call it using `curl`.
+
+```sh
+# call with just the cacert, should get the message "blocked access to mTLS-protected resource"
+curl -v https://$VM_IP --cacert cert.pem
+
+# now call with cert and key, should get the message "access to mTLS-protected resource"
+curl -v https://$VM_IP --cacert cert.pem --key key.pem --cert cert.pem
+```
+
+## Step 6: Create Apigee Keystore for cert and key to use in a proxy
+
+```sh
+# create Apigee keystore
+apigeecli keystores create -n mtls-keystore1 -e $APIGEE_ENV -o $PROJECT_ID -t $(gcloud auth print-access-token)
```

Comment:
Need to add a Pre-req that says you need to insall gcloud, apigeecli
also mention the roles needed - to create proxy, VM, create firewall, etc